Code/Framework/AzCore/AzCore/Name/NameDictionary.h

```diff
@@ -77,6 +77,8 @@ namespace AZ
         //! @return A Name instance. If the hash was not found, the Name will be empty.
         Name FindName(Name::Hash hash) const;
 
+        void LoadDeferredNames(Name* deferredHead);
```

Comment:
Comment please, especially state this is really only intended for internal use.

Reply:
Alternatively this can be made private and the NameDictionary can friend the extern "C" `CreateModuleClass` function

Reply:
I couldn't figure out an easy way to make this work with `friend` (`CreateModuleClass` seems to vary in signature) but I can take another look or add a method to module (which can be friends with `Name`) if desired - it's documented as non-public now at least.

Reply:
Did you try `friend ::CreateModuleClass(void*)`?

Reply:
I tried `friend AZ::Module* ::CreateModuleClass();` which needs a forward declaration for `Module` which I got MSVC issues with because I didn't always forward declare with the right DLL linkage specifier - I could friend `Module` itself and add something there, but that feels a bit iffy to me vs just marking this as internal-only API

Reply:
CreateModuleClass accepts a parameter of `(void*)` not blank `()`

Have you tried forward declaring `extern "C" void CreateModuleClass(void*);`?

Here is how you would forward declare it https://godbolt.org/z/YG39b36oG

Reply:
I have, yeah - what bit me was actually the warning about mismatched DLL import/export, e.g.:
https://godbolt.org/z/joohzhjs4